### PR TITLE
Home Screen: Get medications for the selected date from DB instead of filtering

### LINKED ISSUE #114: Home Screen: Get medications for the selected date from DB instead of filtering

### PR DESCRIPTION
**Pull request by daniel-waiguru**
This pull request contains the following changes:
* Home Screen: Get medications for the selected date from DB instead of filtering: Issue #114 

Please review the changes and let me know if you have any questions.

Closes #114

### CODE REVIEW

File: app/src/main/java/com/waseefakhtar/doseapp/extension/DateExtension.kt [hunk 1]
```diff
@@ -15,6 +15,21 @@ fun Date.toFormattedMonthDateString(): String {
     return sdf.format(this)
 }
 
+fun Date.toFormattedYeahMonthDateString(): String {
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/extension/DateExtension.kt:18]:
```suggestion
fun Date.toFormattedYearMonthDateString(): String {
```

File: app/src/main/java/com/waseefakhtar/doseapp/feature/home/viewmodel/HomeViewModel.kt [hunk 2]
```diff
@@ -56,4 +80,7 @@ class HomeViewModel @Inject constructor(
     fun getUserPlan() {
         // TODO: Get user plan
     }
+    companion object {
+        const val DATE_FILTER_KEY = "com.waseefakhtar.doseapp.MedicationDateFilter"
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/feature/home/viewmodel/HomeViewModel.kt:84]: Why don't we go for a simple `medication_date_filter` instead? :)

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -7,7 +7,7 @@ version-name = "1.4.1"
 accompanistPermissions = "0.33.2-alpha"
 activity = "1.8.0"
 androidxHilt = "1.1.0"
-compose = "1.6.0-alpha08"
+compose = "1.6.0-beta03"
```
Comment (waseefakhtar) [inline gradle/libs.versions.toml:10]: Great that you cared about updating from the alphas! :) However, I'd recommend to keep updating dependencies to a separate PR so they're easily traceable in case any changes from the latest dependencies breaks the app.

**Review by waseefakhtar — CHANGES_REQUESTED** [review]
Looks pretty good to me! 🚀 I just had some minor comments.

File: app/src/main/java/com/waseefakhtar/doseapp/feature/home/viewmodel/HomeViewModel.kt (hunk 2 above)
Comment (daniel-waiguru) [reply app/src/main/java/com/waseefakhtar/doseapp/feature/home/viewmodel/HomeViewModel.kt:84]: You mean the key string?

File: gradle/libs.versions.toml (hunk 3 above)
Comment (daniel-waiguru) [reply gradle/libs.versions.toml:10]: Makes sense, should I revert this?

File: app/src/main/java/com/waseefakhtar/doseapp/extension/DateExtension.kt (hunk 1 above)
Comment (daniel-waiguru) [reply app/src/main/java/com/waseefakhtar/doseapp/extension/DateExtension.kt:18]: What about this? I don't see the difference

Comment (ikerfah) [reply app/src/main/java/com/waseefakhtar/doseapp/extension/DateExtension.kt:18]: Yeah -> Year

**Comment by daniel-waiguru** [comment]
> Looks pretty good to me! 🚀 I just had some minor comments.

Addressed

**Review by waseefakhtar — APPROVED** [review]
Looks great @daniel-waiguru! 👏 Sorry I couldn't review this sooner but thank you for your contribution! :)
